src/form/widget/widget.js

```diff
@@ -0,0 +1,379 @@
+     * @param {String} method - The name of the method to call.
+     * @param {...} args - Extra parameters
+     * @returns {*}
+     */
+    const delegate = (method, ...args) => {
+        if (_.isFunction(provider[method])) {
+            return provider[method].apply(widget, args);
+        }
+        return defaultProvider[method].apply(widget, args);
+    };
+
+    /**
+     * @typedef {component} widgetForm
+     */
+    const widgetApi = {
+        /**
+         * Gets the widget's URI
+         * @returns {String}
+         */
+        getUri() {
+            return this.getConfig().uri;
+        },
+
+        /**
+         * Gets the value of the widget
+         * @returns {String|String[]}
+         */
+        getValue() {
+            return delegate('getValue');
+        },
+
+        /**
+         * Sets the value of the widget
+         * @param {String|String[]} value
+         * @returns {widgetForm}
+         * @fires change after the value has been changed
+         */
+        setValue(value) {
+            this.getConfig().value = value;
+            delegate('setValue', value);
+            this.notify();
+
+            return this;
+        },
+
+        /**
+         * Gets access to the validation engine
+         * @returns {validator}
+         */
+        getValidator() {
+            return validator;
+        },
+
+        /**
+         * Sets the validation engine
+         * @param {validationRule|validationRule[]|validator} validation
+         * @returns {widgetForm}
+         */
+        setValidator(validation) {
+            if (validation && _.isFunction(validation.validate)) {
+                validator = validation;
+            } else {
+                if (validation && !validation.validations) {
+                    if (!_.isArray(validation)) {
+                        validation = [validation];
+                    }
+                    validation = {
+                        validations: validation
+                    };
+                }
+
+                validator = validatorFactory(validation);
+            }
+
+            return this;
+        },
+
+        /**
+         * Resets the widget to the default validators
+         * @returns {widgetForm}
+         */
+        setDefaultValidators() {
+            // restore factory default validators
+            this.setValidator(this.getConfig().validator);
+
+            // then apply provider default validators
+            delegate('setDefaultValidators');
+
+            return this;
+        },
+
+        /**
+         * Resets the widget to its default value
+         * @returns {widgetForm}
+         */
+        reset() {
+            delegate('reset');
+            setInvalidState(false);
+            return this;
+        },
+
+        /**
+         * Serializes the value of the widget
+         * @returns {widgetValue}
+         */
+        serialize() {
```

Comment:
I don't like the `serialize` name, it is misleading.This method role is to get the widget values and should have been called  `getValues`, no ? 
You don't need to change it but this can be an improvement for the next time.

Reply:
I will replace it by `serializeValue()`, will do the change in #17